azure-pipelines.yml

```diff
@@ -291,3 +291,64 @@ jobs:
       twine upload -u giotto-learn -p $(pypi_psw) --skip-existing dist/*
     condition: eq(variables['nightly_check'], 'true')
     displayName: 'Upload nightly wheels to PyPI'
+
+- job: 'Documentation'
+  pool:
+    vmImage: 'macOS-10.14'
+  strategy:
+    matrix:
+      Python38:
+        python.version: '3.8'
+  variables:
+    EXECUTE: 'true'
+    RUNNOTEBOOKS: TRUE
+    VERSION: master
+
+  steps:
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(python.version)'
+
+    - script: |
+        set -e
+        brew update
+        brew install boost
+      displayName: 'Install system dependencies'
+
+    - script: |
+        set -e
+        python -m pip install --upgrade pip setuptools
+        python -m pip install -e ".[dev,doc]"
+      displayName: 'Install dependencies and dev environment'
+
+    - script: |
+        set -e
+        cd doc/
+        make html
+      displayName: 'Build docs'
+
+    - script: |
+        ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+      displayName: 'Create .ssh/known_hosts'
+      condition: ne(variables['Build.Reason'], 'PullRequest')
+
+    - task: InstallSSHKey@0
+      inputs:
+        knownHostsEntry: ~/.ssh/known_hosts
+        sshPublicKey: $(KEY_GTDA-DOCS_PUB)
+        sshKeySecureFile: KEY_GTDA-DOCS
+      condition: ne(variables['Build.Reason'], 'PullRequest')
+
+    - script: |
+        set -e
+        cd /tmp/
+        git clone git@github.com:giotto-learn/gtda-docs.git
+        cd gtda-docs
+        git config user.email "maintainers@giotto.ai"
+        git config user.name "giotto-maintainers"
+        # TODO: copy docs to the right folder and git add them
+        git commit -m "Test empty commit (azure pipelines)" --allow-empty
+        git push -u origin master
+        cd -
+      displayName:  'Commit the docs'
+      condition: ne(variables['Build.Reason'], 'PullRequest')
```

Comment:
@ulupo , I don't recall what we agreed on for triggering changes in the documentation - sorry. Would you mind recalling?